compiler/rustc_lint/src/unused.rs

```diff
@@ -1340,7 +1340,33 @@ declare_lint! {
     "unnecessary braces around an expression"
 }
 
-declare_lint_pass!(UnusedBraces => [UNUSED_BRACES]);
+#[derive(Default)]
+pub(crate) struct UnusedBraces {
+    parent_followed_by_block: Vec<bool>,
```

Comment:
```suggestion
    followed_by_block: Vec<bool>,
```
and please add a short comment why this is used